timesketch/models/acl.py

```diff
@@ -0,0 +1,182 @@
+# Copyright 2015 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This module implements the Access Control List (ACL) system.
+
+The ACL systems is simple. It works by creating a many-to-many relationship
+to the model that needs ACL functionality. This is implemented as a MixIn to
+make it easy to annotate models to give them access to the ACL system.
+
+The model has the following permissionsL: "read", "write" and "delete".
+"""
+
+from flask_login import current_user
+from sqlalchemy import Column
+from sqlalchemy import ForeignKey
+from sqlalchemy import Integer
+from sqlalchemy import or_
+from sqlalchemy import not_
+from sqlalchemy import String
+from sqlalchemy.ext.declarative import declared_attr
+from sqlalchemy.orm import relationship
+
+from timesketch.models import BaseModel
+from timesketch.models import db_session
+
+
+class AccessControlEntry(object):
+    """
+    Access Control Entry database model. It has a user object (instance of
+    timesketch.models.user.User) and a permission (read, write or delete).
+    """
+    @declared_attr
+    def user_id(self):
+        """Foreign key to a user model.
+
+        Returns:
+            A column (instance of sqlalchemy.Column)
+        """
+        return Column(Integer, ForeignKey('user.id'))
+
+    @declared_attr
+    def user(self):
+        """A relationship to a user object.
+
+        Returns:
+            A relationship (instance of sqlalchemy.orm.relationship)
+        """
+        return relationship('User')
+
+    # Permission column (read, write or delete)
+    permission = Column(String(255))
```

Comment:
This is SQLAlchemy column type for creating the SQL schema. Every string field in the schema uses this, not only permissions.